localstack/utils/aws/templating.py

```diff
@@ -67,9 +85,8 @@ def apply(obj, **_):
                         obj[k] = ExtendedString(v)
             return obj
 
-        # loop through the variables and enable certain additional util
-        # functions (e.g., string utils)
-        variables = copy.deepcopy(variables or {})
+        # loop through the variables and enable certain additional util functions (e.g., string utils)
+        variables = {} if variables is None else variables
```

Comment:
Is copying not necessary here? What side effects does removing `deepcopy` cause here?

Reply:
That's a great catch. This change is required to allow in-place modification of elements in VTL.

For example, in AppSync we have mutable elements like the `stash` which we want to be able to modify like this in resolver mapping templates:
```
$ctx.stash.put("email", $ctx.args.input.email)
```

The ability to do in-place modification is covered by a small unit test in this PR (`test_put_value_to_dict(..)`)